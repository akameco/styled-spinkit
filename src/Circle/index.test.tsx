import * as React from 'react'
import renderer from 'react-test-renderer'
import Comp from '.'

test('render without props', () => {
  const tree = renderer.create(<Comp />).toJSON()
  expect(tree).toMatchSnapshot()
})
