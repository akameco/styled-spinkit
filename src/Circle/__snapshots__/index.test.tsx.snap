// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without props 1`] = `
.c2 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: rotate(30deg);
  -ms-transform: rotate(30deg);
  transform: rotate(30deg);
}

.c2::before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  border-radius: 100%;
  -webkit-animation: fQdAWJ 1.2s ease-in-out -1.2s infinite both;
  animation: fQdAWJ 1.2s ease-in-out -1.2s infinite both;
}

.c3 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: rotate(60deg);
  -ms-transform: rotate(60deg);
  transform: rotate(60deg);
}

.c3::before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  border-radius: 100%;
  -webkit-animation: fQdAWJ 1.2s ease-in-out -1.1s infinite both;
  animation: fQdAWJ 1.2s ease-in-out -1.1s infinite both;
}

.c4 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c4::before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  border-radius: 100%;
  -webkit-animation: fQdAWJ 1.2s ease-in-out -1s infinite both;
  animation: fQdAWJ 1.2s ease-in-out -1s infinite both;
}

.c5 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: rotate(120deg);
  -ms-transform: rotate(120deg);
  transform: rotate(120deg);
}

.c5::before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  border-radius: 100%;
  -webkit-animation: fQdAWJ 1.2s ease-in-out -0.9s infinite both;
  animation: fQdAWJ 1.2s ease-in-out -0.9s infinite both;
}

.c6 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: rotate(150deg);
  -ms-transform: rotate(150deg);
  transform: rotate(150deg);
}

.c6::before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  border-radius: 100%;
  -webkit-animation: fQdAWJ 1.2s ease-in-out -0.8s infinite both;
  animation: fQdAWJ 1.2s ease-in-out -0.8s infinite both;
}

.c7 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c7::before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  border-radius: 100%;
  -webkit-animation: fQdAWJ 1.2s ease-in-out -0.7s infinite both;
  animation: fQdAWJ 1.2s ease-in-out -0.7s infinite both;
}

.c8 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: rotate(210deg);
  -ms-transform: rotate(210deg);
  transform: rotate(210deg);
}

.c8::before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  border-radius: 100%;
  -webkit-animation: fQdAWJ 1.2s ease-in-out -0.6s infinite both;
  animation: fQdAWJ 1.2s ease-in-out -0.6s infinite both;
}

.c9 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: rotate(240deg);
  -ms-transform: rotate(240deg);
  transform: rotate(240deg);
}

.c9::before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  border-radius: 100%;
  -webkit-animation: fQdAWJ 1.2s ease-in-out -0.5s infinite both;
  animation: fQdAWJ 1.2s ease-in-out -0.5s infinite both;
}

.c10 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
}

.c10::before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  border-radius: 100%;
  -webkit-animation: fQdAWJ 1.2s ease-in-out -0.4s infinite both;
  animation: fQdAWJ 1.2s ease-in-out -0.4s infinite both;
}

.c11 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: rotate(300deg);
  -ms-transform: rotate(300deg);
  transform: rotate(300deg);
}

.c11::before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  border-radius: 100%;
  -webkit-animation: fQdAWJ 1.2s ease-in-out -0.3s infinite both;
  animation: fQdAWJ 1.2s ease-in-out -0.3s infinite both;
}

.c12 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: rotate(330deg);
  -ms-transform: rotate(330deg);
  transform: rotate(330deg);
}

.c12::before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  border-radius: 100%;
  -webkit-animation: fQdAWJ 1.2s ease-in-out -0.2s infinite both;
  animation: fQdAWJ 1.2s ease-in-out -0.2s infinite both;
}

.c13 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: rotate(360deg);
  -ms-transform: rotate(360deg);
  transform: rotate(360deg);
}

.c13::before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  border-radius: 100%;
  -webkit-animation: fQdAWJ 1.2s ease-in-out -0.1s infinite both;
  animation: fQdAWJ 1.2s ease-in-out -0.1s infinite both;
}

.c0 {
  margin: 40px auto;
  width: 40px;
  height: 40px;
  position: relative;
}

.c0 > .c1::before {
  background-color: #333;
}

<div
  className="c0"
  color="#333"
  size={40}
>
  <div
    className="c1 c2"
    transform={30}
  />
  <div
    className="c1 c3"
    transform={60}
  />
  <div
    className="c1 c4"
    transform={90}
  />
  <div
    className="c1 c5"
    transform={120}
  />
  <div
    className="c1 c6"
    transform={150}
  />
  <div
    className="c1 c7"
    transform={180}
  />
  <div
    className="c1 c8"
    transform={210}
  />
  <div
    className="c1 c9"
    transform={240}
  />
  <div
    className="c1 c10"
    transform={270}
  />
  <div
    className="c1 c11"
    transform={300}
  />
  <div
    className="c1 c12"
    transform={330}
  />
  <div
    className="c1 c13"
    transform={360}
  />
</div>
`;
